pkg/client/unversioned/request.go

```diff
@@ -149,6 +150,10 @@ func (r *Request) Resource(resource string) *Request {
 		r.err = fmt.Errorf("resource already set to %q, cannot change to %q", r.resource, resource)
 		return r
 	}
+	if ok, msg := validation.ValidatePathSegmentName(resource, false); !ok {
```

Comment:
a name like ".." won't make it to the server intact... where else would you suggest we do it?